# SVM : 서포트 벡터 머신

# AND, OR, XOR

### 1. AND 연산 : 두 개의 값이 모두 참일 때 참이된다

### 2. OR 연산 : 둘 중에 하나만 참이어도 참이된다.

### 3. XOR 연산 : 두 개의 값이 다른 경우 참이된다. 

## XOR()

| A | B | Result |
|-|-|-|
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | 0 | 1 |
| 1 | 1 | 0 |

In [ ]:
from sklearn import svm

# 데이터 전처리 : 학습용/테스트 데이터 분리

# 분류작업 머신
clf = svm.SVC()

    # train_data, train_label
clf.fit(학습데이터, 학습정답)

    # test_data
predict(테스트데이터)

            # test_label, 예측값
accuracy_score(테스트정답, 예측값<predict>)

## 실제코드

In [7]:
# AND 연산
from sklearn import svm
from sklearn.metrics import accuracy_score

clf = svm.SVC()

clf.fit([
    [0, 0],
    [1, 0],
    [0, 1],
    [1, 1]
], 
    [0, 0, 0, 1]
)

predict = clf.predict([
    [1, 0],
    [1, 1]
])

print("예측결과 : ", predict)

score = accuracy_score([[0],[1]], predict)

print("정확도 : " , score)

예측결과 :  [0 1]
정확도 :  1.0


In [9]:
from sklearn import svm

train_data = [[0, 0], [0, 1], [1, 0], [1, 1]]
train_label = [0, 0, 0, 1]

test_data = [[1, 0], [1, 1]]
test_label = [0, 1]

clf = svm.SVC()

clf.fit(train_data, train_label)

predict = clf.predict(test_data)
print("예측결과 : " , predict)

score = accuracy_score(test_label, predict)
print("정확도 : " , score)

예측결과 :  [0 1]

정확도 :  1.0


# 사이킷런 내장 데이터 사용하기

In [10]:
# 사이킷런을 설치하면 샘플용 데이터가 자동으로 설치된다.
# from sklearn.datasets import 탭키를 누르면 샘플데이터를 확인할 수 있다.

import pandas as pd
from sklearn.datasets import load_iris

In [11]:
# 데이터 객체 생성
dataset = load_iris()

In [12]:
# dataset에 있는 key 값을 확인 (열 이름)
dataset.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

In [ ]:
# dataset에 있는 values 값을 확인 (데이터)
dataset.values()

### 내장된 데이터의 정답 확인 : target

In [14]:
dataset['target']

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2])

In [15]:
dataset['target_names']

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

- 상세 정보 확인

In [17]:
print(dataset['DESCR'])

.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

    :Number of Instances: 150 (50 in each of three classes)
    :Number of Attributes: 4 numeric, predictive attributes and the class
    :Attribute Information:
        - sepal length in cm
        - sepal width in cm
        - petal length in cm
        - petal width in cm
        - class:
                - Iris-Setosa
                - Iris-Versicolour
                - Iris-Virginica
                
    :Summary Statistics:

    ============== ==== ==== ======= ===== ====================
                    Min  Max   Mean    SD   Class Correlation
    ============== ==== ==== ======= ===== ====================
    sepal length:   4.3  7.9   5.84   0.83    0.7826
    sepal width:    2.0  4.4   3.05   0.43   -0.4194
    petal length:   1.0  6.9   3.76   1.76    0.9490  (high!)
    petal width:    0.1  2.5   1.20   0.76    0.9565  (high!)
    ============== ==== ==== ======= ===== ===========

- 최적의 결과를 위해 데이터를 무작위로 섞는다.

In [ ]:
# 수동으로 작업하는 경우

import random
random.shuffle(데이터)

# 자동으로 작업하는 경우

from sklearn.model_selection import train_test_split
train_data, test_data, train_label, test_label = 
                        train_test_split(dataset['data'], dataset['target'])

In [27]:
from sklearn.model_selection import train_test_split

train_data, test_data, train_label, test_label = train_test_split(dataset['data'],dataset['target'])

In [21]:
print(train_data.shape)
print(test_data.shape)
print(train_label.shape)
print(test_label.shape)

(112, 4)
(38, 4)
(112,)
(38,)


In [31]:
from sklearn import svm

clf = svm.SVC()

clf.fit(train_data, train_label)

predict = clf.predict(test_data)
print("예측결과 : " , predict)

score = accuracy_score(test_label, predict)
print("정확도 : " , score)

예측결과 :  [1 0 0 0 1 2 2 2 1 2 2 0 2 2 1 2 0 1 0 1 0 1 0 2 1 2 2 0 2 2 1 2 0 0 2 1 1
 0]
정확도 :  0.9473684210526315


In [34]:
from sklearn import metrics

report = metrics.classification_report(test_label, predict)
print(report)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        12
           1       0.91      0.91      0.91        11
           2       0.93      0.93      0.93        15

    accuracy                           0.95        38
   macro avg       0.95      0.95      0.95        38
weighted avg       0.95      0.95      0.95        38



precision : 예측

0은 100%로 예측을 했다. 그의 신뢰도(recall)도 100%
1은 91프로로 예측을 했다. 그의 신뢰도도 91프로이다.

# one-hot-incoding

In [ ]:
import pandas as pd
from sklearn import svm
from sklearn.datasets import load_breast_cancer
from sklearn.datasets import load_boston

dataset_cancer = load_breast_cancer()
dataset_boston = load_boston()


'''
clf = svm.SVC()

clf.fit(train_data, train_label)

predict = clf.predict(test_data)
print("예측결과 : " , predict)

score = accuracy_score(test_label, predict)
print("정확도 : " , score)
'''

# 손글씨 데이터 분류 : MNIST

# http://yann.lecun.com/exdb/mnist

- train-images-idx3-ubyte     --- 학습용
- train-labels-idx1-ubyte.gz  ---  정답
- t10k-images-idx3-ubyte.gz   --- 테스트
- t10k-labels-idx1-ubyte.gz   --- 정답

In [38]:
import urllib.request as req
import gzip, os, os.path

savepath = './mnist'
baseurl = 'http://yann.lecun.com/exdb/mnist'
files = [
    "train-images-idx3-ubyte.gz",
    "train-labels-idx1-ubyte.gz",
    "t10k-images-idx3-ubyte.gz",
    "t10k-labels-idx1-ubyte.gz"
]

# 파일 다운로드
if not os.path.exists(savepath):
    os.mkdir(savepath)
    
for f in files:
    url = baseurl + '/' + f
    loc = savepath + '/' + f
    print("download : ", url)
    
    if not os.path.exists(loc):
        req.urlretrieve(url, loc)
        
        
# 압축해제
for f in files:
    gz_file = savepath + '/' + f
    raw_file = savepath + '/' + f.replace('.gz', '')
    print('gzip : ', f)
    
    with gzip.open(gz_file, 'rb') as fp:
        body = fp.read()
        
        with open(raw_file, 'wb') as wp:
            wp.write(body)
            
print("Done")

download :  http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
download :  http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz
download :  http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz
download :  http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz
gzip train-images-idx3-ubyte.gz
gzip train-labels-idx1-ubyte.gz
gzip t10k-images-idx3-ubyte.gz
gzip t10k-labels-idx1-ubyte.gz
Done


# 이미지를 CSV 파일로 변환하기

In [ ]:
import struct

def to_csv(name, maxdata) :
    lbl_f = open('./mnist/' + name + '-labels-idx1-ubyte', 'rb')
    img_f = open('./mnist/' + name + '-images-idx3-ubyte', 'rb')
    csv_f = open('./mnist/' + name + '.csv', 'w', encoding='utf-8')

In [39]:
import struct
def to_csv(name, maxdata):
    # 레이블 파일과 이미지 파일 열기
    lbl_f = open("./mnist/"+name+"-labels-idx1-ubyte", "rb")
    img_f = open("./mnist/"+name+"-images-idx3-ubyte", "rb")
    csv_f = open("./mnist/"+name+".csv", "w", encoding="utf-8")
    
    # 헤더 정보 읽기 --- (※1)
    # unpack : 데이터를 풀겠다.
    # >II : 부호없는 정수형을 가져오겠다.
    mag, lbl_count = struct.unpack(">II", lbl_f.read(8))
    mag, img_count = struct.unpack(">II", img_f.read(8))
    rows, cols = struct.unpack(">II", img_f.read(8))
    pixels = rows * cols
    # 28 * 28
    
    # 이미지 데이터를 읽고 CSV로 저장하기 --- (※2)
    res = []
    for idx in range(lbl_count):
        if idx > maxdata: break
        label = struct.unpack("B", lbl_f.read(1))[0]
        bdata = img_f.read(pixels)
        sdata = list(map(lambda n: str(n), bdata))
        csv_f.write(str(label)+",")
        csv_f.write(",".join(sdata)+"\r\n")
        
        # 잘 저장됐는지 이미지 파일로 저장해서 테스트하기 -- (※3)
        if idx < 10:
            s = "P2 28 28 255\n"
            s += " ".join(sdata)
            iname = "./mnist/{0}-{1}-{2}.pgm".format(name,idx,label)
            with open(iname, "w", encoding="utf-8") as f:
                f.write(s)
    csv_f.close()
    lbl_f.close()
    img_f.close()
    
# 결과를 파일로 출력하기 --- (※4)
to_csv("train", 60000)
to_csv("t10k", 10000)


### 데이터를 저장하는 방식 : 12345678 데이터가 있다고 가정

#### 리틀엔디안 방식으로 저장 : 역순으로 저장

- 0x78 -> 0x56 -> 0x32 -> 0x12

#### 빅엔디안 방식으로 저장 : 순서대로 저장

- 0x12 -> 0x34 -> 0x56 -> 0x78

In [42]:
%%writefile number.py

input = '5 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 18 18 18 126 136 175 26 166 255 247 127 0 0 0 0 0 0 0 0 0 0 0 0 30 36 94 154 170 253 253 253 253 253 225 172 253 242 195 64 0 0 0 0 0 0 0 0 0 0 0 49 238 253 253 253 253 253 253 253 253 251 93 82 82 56 39 0 0 0 0 0 0 0 0 0 0 0 0 18 219 253 253 253 253 253 198 182 247 241 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 80 156 107 253 253 205 11 0 43 154 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 14 1 154 253 90 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 139 253 190 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 11 190 253 70 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 35 241 225 160 108 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 81 240 253 253 119 25 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 45 186 253 253 150 27 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 16 93 252 253 187 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 249 253 249 64 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 46 130 183 253 253 207 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 39 148 229 253 253 253 250 182 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 24 114 221 253 253 253 253 201 78 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 23 66 213 253 253 253 253 198 81 2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 18 171 219 253 253 253 253 195 80 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 55 172 226 253 253 253 253 244 133 11 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 136 253 253 253 212 135 132 16 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'
input = input.split(' ') # 공백기준으로 데이터 분리

for i in range(len(input)):
    print('{:3}'.format(input[i]), end=' ')
    # print() 함수는 괄호를 닫을 때 Enter한다. end연산자는 한 줄로 만들때 사용
    # end : 모든 데이터를 한 줄로 처리하라는 의미.
    
    if i % 28 == 0:
        print()

Overwriting number.py


# 데이터 분리 (학습용/테스트용)

In [43]:
# header=None 이 값을 생략할 경우 첫번째 행이 헤더로 선언된다.

train_csv = pd.read_csv('./mnist/train.csv', header=None)
tk_csv = pd.read_csv('./mnist/t10k.csv', header=None)

In [44]:
len(train_csv)

60000

In [45]:
len(tk_csv)

10000

In [55]:
train_label = train_csv[0].values
test_label = tk_csv[0].values

In [56]:
train_data = train_csv.iloc[:, 1:].values
test_data = tk_csv.iloc[:, 1:].values

In [57]:
print(len(train_label))
print(len(train_data))
print(len(test_label))
print(len(test_data))

60000
60000
10000
10000


loc[A4] : 하나만 가져올 때
iloc[시작행:마지막행, 시작열:마지막열] : 여러 데이터 가져올 때

In [58]:
from sklearn import svm

clf = svm.SVC()

clf.fit(train_data, train_label)

predict = clf.predict(test_data)
print("예측결과 : " , predict)

score = accuracy_score(test_label, predict)
print("정확도 : " , score)

예측결과 :  [7 2 1 ... 4 5 6]
정확도 :  0.9792


In [59]:
report = metrics.classification_report(test_label, predict)
print(report)

              precision    recall  f1-score   support

           0       0.98      0.99      0.99       980
           1       0.99      0.99      0.99      1135
           2       0.98      0.97      0.98      1032
           3       0.97      0.99      0.98      1010
           4       0.98      0.98      0.98       982
           5       0.99      0.98      0.98       892
           6       0.99      0.99      0.99       958
           7       0.98      0.97      0.97      1028
           8       0.97      0.98      0.97       974
           9       0.97      0.96      0.97      1009

    accuracy                           0.98     10000
   macro avg       0.98      0.98      0.98     10000
weighted avg       0.98      0.98      0.98     10000



# 외국어 문장 판독하는 프로그램

- lang
    - train : 학습용으로 5개국어 샘플 노래 (20개)
    - test : 테스트용으로 사용할 각 언어 샘플 2개씩
    - sample : 영어노래 1개, 불어노래 1개

In [62]:
from sklearn import svm, metrics
import glob, os.path, re, json

# glob 모듈은 파일이 있는 경로를 알려준다
files = glob.glob('./lang/train/*.txt')

#### 정답 추출

- en-1.txt

파일명에 - 기호를 중심으로 split 한다.

In [66]:
from sklearn import svm, metrics
import glob, os.path, re, json

# glob 모듈은 파일이 있는 경로를 알려준다
files = glob.glob('./lang/train/*.txt')

for file_name in files:
    basename = os.path.basename(file_name)
    lang = basename.split('-')[0]
    print(basename, lang)

tl-20.txt tl
en-5.txt en
en-4.txt en
en-1.txt en
en-3.txt en
fr-8.txt fr
tl-18.txt tl
fr-10.txt fr
tl-19.txt tl
fr-9.txt fr
en-2.txt en
id-15.txt id
id-14.txt id
id-13.txt id
id-12.txt id
id-11.txt id
fr-7.txt fr
tl-17.txt tl
tl-16.txt tl
fr-6.txt fr


### 파일들의 데이터를 읽어들인다. (알파벳 빈도수)

In [ ]:
from sklearn import svm, metrics
import glob, os.path, re, json

# glob 모듈은 파일이 있는 경로를 알려준다
files = glob.glob('./lang/train/*.txt')

for file_name in files:
    basename = os.path.basename(file_name)
    lang = basename.split('-')[0]
    
    # 텍스트 추출
    file = open(file_name, 'r', encoding='utf-8')
    text = file.read()  # 파일에 들어있는 전체 데이터를 읽어들인다.
    text = text.lower() # 가독성을 위해 소문자 처리
    file.close()  # 파일 객체를 닫아준다.
    
    for c in text: # 알파벳 한글자씩 처리한다. 누적합계를 구하여 통계낸다.
        print(c, end=' ')

#### 문자를 숫자로 변환하는 함수 = ord('문자')

#### 숫자를 문자로 변환하는 함수 = chr(정수)

In [69]:
ord('홍')

54861

In [71]:
chr(54861)

'홍'

#### 알파벳 출현 빈도수 구하기 (소문자 97 ~ 122)

In [73]:
from sklearn import svm, metrics
import glob, os.path, re, json

# glob 모듈은 파일이 있는 경로를 알려준다
files = glob.glob('./lang/train/*.txt')

for file_name in files:
    basename = os.path.basename(file_name)
    lang = basename.split('-')[0]
    
    # 텍스트 추출
    file = open(file_name, 'r', encoding='utf-8')
    text = file.read()  # 파일에 들어있는 전체 데이터를 읽어들인다.
    text = text.lower() # 가독성을 위해 소문자 처리
    file.close()  # 파일 객체를 닫아준다.
    
    code_a = ord('a') # 97
    code_z = ord('z') # 122

    count = [0] * 26

    for c in text:
        code_current = ord(c)

        if code_a <= code_current <= code_z:
            count[code_current - code_a] += 1
        
    print(lang, count)

tl [1469, 165, 142, 269, 466, 42, 240, 49, 548, 17, 107, 367, 173, 834, 491, 224, 13, 286, 472, 335, 202, 35, 42, 3, 246, 22]
en [337, 93, 142, 181, 645, 93, 93, 260, 297, 15, 27, 181, 132, 284, 302, 86, 2, 331, 272, 436, 114, 49, 109, 14, 68, 3]
en [383, 147, 159, 210, 642, 89, 125, 313, 346, 8, 69, 238, 114, 311, 465, 89, 2, 314, 389, 497, 129, 27, 104, 32, 93, 9]
en [349, 59, 210, 212, 484, 72, 88, 201, 340, 8, 25, 247, 121, 356, 412, 76, 0, 357, 282, 370, 119, 45, 65, 3, 92, 2]
en [259, 44, 165, 118, 434, 53, 91, 85, 342, 9, 6, 206, 81, 273, 277, 75, 6, 195, 318, 293, 105, 68, 43, 2, 65, 2]
fr [585, 127, 321, 412, 1266, 105, 109, 116, 690, 35, 15, 500, 234, 596, 453, 189, 47, 661, 536, 492, 344, 122, 6, 41, 40, 16]
tl [4100, 376, 45, 228, 532, 5, 1632, 201, 1256, 32, 557, 880, 628, 2414, 764, 559, 1, 480, 845, 1064, 621, 19, 255, 1, 352, 1]
fr [1228, 163, 581, 831, 2347, 253, 167, 242, 1082, 36, 64, 1096, 418, 1126, 767, 406, 92, 1171, 1117, 1210, 834, 223, 11, 61, 57, 13]
tl [2054

- a와 z 의 격차가 크기 때문에 표준편차가 크다.
- 따라서 정규화 작업을 해준다. 

en [337, 93, 142, 181, 645, 93, 93, 260, 297, 15, 27, 181, 132, 284, 302, 86, 2, 331, 272, 436, 114, 49, 109, 14, 68, 3]

- 정규화 (Normalization) : 데이터를 0 ~ 1.0 사이의 값으로 대체한다.
    - 모든 데이터를 합산한 후 고유값으로 나눈다.
    
- 머신러닝에서 필요한 값
    - 수치값은 정규화 되어야 한다.

### 필수로 알아야 할 함수

- zip, map, filter, lambda, reduce, 컴퓨리핸션

In [79]:
from sklearn import svm, metrics
import glob, os.path, re, json

# glob 모듈은 파일이 있는 경로를 알려준다
files = glob.glob('./lang/train/*.txt')

train_label = []
train_data = []

for file_name in files:
    basename = os.path.basename(file_name)
    lang = basename.split('-')[0]
    
    # 텍스트 추출
    file = open(file_name, 'r', encoding='utf-8')
    text = file.read()  # 파일에 들어있는 전체 데이터를 읽어들인다.
    text = text.lower() # 가독성을 위해 소문자 처리
    file.close()  # 파일 객체를 닫아준다.
    
    code_a = ord('a') # 97
    code_z = ord('z') # 122

    count = [0] * 26

    for c in text:
        code_current = ord(c)

        if code_a <= code_current <= code_z:
            count[code_current - code_a] += 1
        

    # 정규화
    total = sum(count)
    count = list(map(lambda n: n/total, count))
    
    # 결과 저장
    train_label.append(lang)
    train_data.append(count)
    
# ------------------------------------------------------------------------

# glob 모듈은 파일이 있는 경로를 알려준다
files = glob.glob('./lang/test/*.txt')

test_label = []
test_data = []

for file_name in files:
    basename = os.path.basename(file_name)
    lang = basename.split('-')[0]
    
    # 텍스트 추출
    file = open(file_name, 'r', encoding='utf-8')
    text = file.read()  # 파일에 들어있는 전체 데이터를 읽어들인다.
    text = text.lower() # 가독성을 위해 소문자 처리
    file.close()  # 파일 객체를 닫아준다.
    
    code_a = ord('a') # 97
    code_z = ord('z') # 122

    count = [0] * 26

    for c in text:
        code_current = ord(c)

        if code_a <= code_current <= code_z:
            count[code_current - code_a] += 1
        

    # 정규화
    total = sum(count)
    count = list(map(lambda n: n/total, count))
    
    # 결과 저장
    test_label.append(lang)
    test_data.append(count)

In [80]:
from sklearn import svm

clf = svm.SVC()

clf.fit(train_data, train_label)

predict = clf.predict(test_data)
print("예측결과 : " , predict)

score = accuracy_score(test_label, predict)
print("정확도 : " , score)

report = metrics.classification_report(test_label, predict)
print(report)

예측결과 :  ['tl' 'fr' 'en' 'en' 'en' 'id' 'fr' 'id' 'id' 'fr' 'tl']
정확도 :  1.0
              precision    recall  f1-score   support

          en       1.00      1.00      1.00         3
          fr       1.00      1.00      1.00         3
          id       1.00      1.00      1.00         3
          tl       1.00      1.00      1.00         2

    accuracy                           1.00        11
   macro avg       1.00      1.00      1.00        11
weighted avg       1.00      1.00      1.00        11

